lib/coney/rabbit_connection.ex

```diff
@@ -40,57 +40,71 @@ defmodule Coney.RabbitConnection do
 
     Basic.qos(chan, prefetch_count: connection.prefetch_count)
 
-    queue = declare_queue(chan, connection.queue)
-
-    exchange_name = declare_exchange(chan, Map.get(connection, :exchange, nil))
+    {:ok, _consumer_tag} = Basic.consume(chan, connection.queue, consumer_pid)
+  end
 
-    bind_queue(chan, exchange_name, queue, Map.get(connection, :binding, []))
+  def publish(conn, exchange_name, routing_key, message) do
+    chan = create_channel(conn)
 
-    {:ok, _consumer_tag} = Basic.consume(chan, queue, consumer_pid)
+    try do
+      Basic.publish(chan, exchange_name, routing_key, message)
+    after
+      Channel.close(chan)
+    end
   end
 
-  defp declare_queue(chan, {name}) do
-    declare_queue(chan, {name, []})
+  def confirm(channel, tag) do
+    Basic.ack(channel, tag)
   end
 
-  defp declare_queue(chan, {name, params}) do
-    Queue.declare(chan, name, params)
-    name
+  def reject(channel, tag, opts) do
+    Basic.reject(channel, tag, opts)
   end
 
-  defp declare_exchange(chan, {type, name}) do
-    declare_exchange(chan, {type, name, []})
+  def init_topology(conn, %{exchanges: exchanges, queues: queues}) do
+    channel = create_channel(conn)
+
+    Enum.each(exchanges, &declare_exchange(channel, &1))
+    Enum.each(queues, &declare_queue(channel, &1))
+
+    Channel.close(channel)
```

Comment:
I think here we should capture possible errors as well, because if config doesnt match current rabbit state (for example if an exchange has different type in config and existing rabbit) here could be an exception

Reply:
Hmm, how about letting it die? Because the topology is essential for coney to work, so there is no point in continuing so let it just die?

Reply:
yes, you are right